GH22 Add encoding to compare file functions

IanHopkinson opened the pull request:
This PR adds an encoding keyword argument to the compare files function. It also fixes some instances where tests failed on Windows.

There is not a test for the new compare files functionality since this would be challenging to arrange. 

Issue is described in #22

mcarans commented:
I see you spotted some errors in our ruff config which I'll need to propagate to all the other libraries. Please can you make the config look like this for consistency:
```
line-length = 79
exclude = ["_version.py"]

[lint]
# List of rules: https://docs.astral.sh/ruff/rules/
select = [
  "E",   # pycodestyle - default
  "F",   # pyflakes - default
  "I"    # isort
]
ignore = [
    "E501" # Line too long
]

[lint.isort]
known-local-folder = ["hdx"]
```
For load_text, I think it would be better if `linesep_` was a parameter which could be called `default_line_separator` with default value "\n". `replace_newlines` would then be renamed `replace_line_separators` for consistency and the documentation updated accordingly.

IanHopkinson commented:
I've updated as requested. I added a `DepreciationWarning` if `replace_newlines` is supplied since potentially this is a breaking change. As it is the only place "replace_newlines" was used was in the test for `load_text`!

mcarans commented on src/hdx/utilities/loader.py:
```diff
@@ -19,24 +19,38 @@ def load_text(
+        replace_newlines (Optional[str]): String with which to replace newlines. Defaults to None (don't replace). (deprecated 2024-02-07)
+        replace_line_separators (Optional[str]): String with which to replace newlines. Defaults to None (don't replace).
         loaderror_if_empty (bool): Whether to raise LoadError if file is empty. Default to True.
+        default_line_separator (str): line separator to be replaced if replace_newlines is True
```
Description needs fixing eg. something like:
default_line_separator (str): line separator to be replaced if replace_line_separators is not None

IanHopkinson replied:
Update made, as requested - I also added a date to the depreciation warning

mcarans reviewed (approved):
Looks great! I've suggested one minor documentation fix and approved.

IanHopkinson commented:
Merged, failing "test" is because of a very small drop in coverage (0.1%). Requires publication to PyPI
